### DISCUSSION TITLE
Windows Terminal Canary (nightly build) is now available!

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#16121] [Announcements]
Hey all! 👋

# <img src="https://github.com/microsoft/terminal/blob/main/res/terminal/images-Can/Square44x44Logo.targetsize-72.png?raw=true" width="36" height="36"> Windows Terminal Canary is here!

While we work on refining our distribution plans for Terminal Canary, I wanted to post some getting-started links here for you adventurous folks who want to live on the leading edge. It tracks our `main` branch and will almost certainly be broken from time to time!

### App Installer

This is an evergreen, architecture-independent application installer that supports automatic updating.

> [!NOTE]
> This installer works on all versions of Windows 11, and Windows 10 after the installation of [KB5037768](https://support.microsoft.com/en-us/topic/may-14-2024-kb5037768-os-builds-19044-4412-and-19045-4412-cb676101-1641-4a6c-b512-8b277606c6e3) (May 14, 2024)

### Portable ZIP

Tried-and-true zip files, with no automatic updates _or automatic checking for updates._

Terminal Canary portable works on Windows 10 (19041+) and Windows 11.

_Learn more about the [types of Windows Terminal distributions](https://learn.microsoft.com/windows/terminal/distributions)._

### Others?

For now, we're not intending to publish Canary channel updates via WinGet or the GitHub releases page, as we feel that they would be much too noisy.

---

| Distribution  | Architecture    | Link                                                 |
|---------------|:---------------:|------------------------------------------------------|
| App Installer | x64, arm64, x86 | [download](https://aka.ms/terminal-canary-installer) |
| Portable ZIP  | x64             | [download](https://aka.ms/terminal-canary-zip-x64)   |
| Portable ZIP  | ARM64           | [download](https://aka.ms/terminal-canary-zip-arm64) |
| Portable ZIP  | x86             | [download](https://aka.ms/terminal-canary-zip-x86)   |

**Comment by DHowett** [comment]
## Dustin's Wishlist

- Display the git commit in the About dialog
- Some way to check for (and install?) updates for the ZIP version
- What if the About dialog showed what Velocity features were enabled? That would be cool
- Somehow tell users what changes to look for
- Make sure that Canary works as "Default Console Host"

**Reply by Dhyfer1** [reply]
Thanks for sharing the link for the canary distribution, but why is there no link to download the dev version?

**Reply by DHowett** [reply]
Canary is the same as Dev, we just give Dev different branding so that people can still work on the project! If we started distributing Dev builds, people wouldn't be able to build and debug Terminal with Visual Studio because of conflicts 😄

**Reply by DHowett** [reply]
If I may ask, why are you hoping to get the Dev branded build specifically?

**Reply by XenoPanther** [reply]
I hope you don't mind me asking, do you have an ETA on getting Canary to work as the default console host?

**Reply by zadjii-msft** [reply]
Canary should work as the default Terminal today (assuming you used the app installer on Windows 11). The default terminal stuff requires an app package, so the zip installs won't be able to work for that unfortunately. 

I don't think we have any plans to just replace the default _console host_ at this time.

**Comment by zcobol** [comment]
How to update terminal canary versions? Checking update there was `1.20.2851.0` available but the only way to upgrade was to download the `*.appinstaller` and double-click on it. Is there any other way to upgrade canary?

**Reply by akulbe** [reply]
When I went to the "About" page, under the version it said "Update available" and then when I restarted it, it was at version 1.20.2861.0
![WT_About_Page](https://github.com/microsoft/terminal/assets/123279/efd6f1b1-e695-46aa-8840-8528cff21c3f)

**Reply by giggio** [reply]
Same thing here. But I had to force close the `WindowsTerminal.exe` process under that was started under `svchost.exe`.
I noticed the version changed. Before:
<img width="150" src="https://github.com/microsoft/terminal/assets/334958/57996346-9c44-403c-a75b-6f5a4eec5bf0" />

After:
<img width="150" src="https://github.com/microsoft/terminal/assets/334958/6c1e9a67-8e09-41ef-a61a-77faf26b72cd" />

~There should be a command to update/check for update, I'll open an issue for that.~
There is already an issue for that: #15132

**Comment by DHowett** [comment]
Hey all! If you're on Windows 10 with the most recent updates, you should now be able to install the MSIX Bundle distribution of Canary. I've updated the body of this announcement to explain that!

**Reply by akulbe** [reply]
Wait.. what? you mean people still run Windows 10? 😂

**Comment by DHowett** [comment]
Quick note: with tonight’s Canary build (1.22.1932.0, available now!) and in future builds we are including the new console hosting rewrite.

If you’re using Canary and see any compatibility issues with console applications, please note them in this discussion thread!

**Comment by blu** [comment]
Just switched from stock 1.20.11781.0 to canary 1.22.2081.0, and the first thing I noticed in WSL 188x61 term was that a 60fps full-screen text animation went from 10-11% cpu load to 5-6% cpu load, on Surface Pro X w/ 23H2. Good job!

**Comment by rhoog-ine** [comment]
Why is the AI response so dark it's unreadable?
Why can I only ask about shell commands?

![image](https://github.com/user-attachments/assets/46fd4245-e09f-4db7-b26a-45a8f93e24f3)

**Reply by lhecker** [reply]
You're encountering issue #18270. I'm not entirely sure what you mean with "Why can I only ask about shell commands?". What question related to terminals isn't working for you?

**Reply by rhoog-ine** [reply]
The prompt entry box says "Ask me anything about Shell commands..." and if I ask "what is the weather like today?" I get "I'm sorry, but I am specifically designed to assist with questions about shell commands. If you have any questions related to shell commands in WSL, I'll be happy to help!"

However, I found this copilot plugin for vim and it seems to provide useful code completion hints. LLM AI is apparently not a 'one size fits all' (yet/anymore).

**Comment by izambe** [comment]
check canary update module 
https://www.powershellgallery.com/packages/TerminalCanaryUpdater/1.0.0

**Comment by denis-komarov** [comment]
Get a nightly build version via PowerShell in one line:
```powershell
( [xml] ( ( Invoke-WebRequest -Uri 'https://terminalbuilds-grbmacf3f6bsbma8.z01.azurefd.net/nightly/Microsoft.WindowsTerminalCanary.appinstaller' ).RawContent -replace '(?s)^.*?(\<\?xml.*)$', '${1}' ) ).AppInstaller.MainBundle.Version -replace '^300'
```
